### PR TITLE
Parsoid storage simplification, store old revision edits in new stash

### PR DESCRIPTION
**Pull request by Pchelolo**
Switching stash for editing older revisions and completely getting rid of using the old stashing buckets.

Technically, we don't need to stash all teh old revisions, only for the purpose of VE, but temporary it's ok.

### CODE REVIEW

File: sys/parsoid.js [hunk 1]
```diff
@@ -219,7 +219,13 @@ class ParsoidService {
         ]);
     }
 
-    getNGBucketURI(rp, format, tid) {
+    getLatestBucketURI(domain, title, revision, tid) {
+        return new URI([
+            domain, 'sys', 'key_value', 'parsoid', `${title}:${revision}:${tid}`
```
Comment (d00rman) [inline sys/parsoid.js]: If this is the latest bucket, why are we storing the title together with the revision and tid? it should just be the title

File: sys/parsoid.js [hunk 2]
```diff
@@ -230,18 +236,28 @@ class ParsoidService {
...
+            uri: this.getOldLatestBucketURI(rp, format, tid)
+        })
+        .catch({ status: 404 }, (e) => {
+            if (rp.revision && tid) {
+                return hyper.get({
+                    uri: this.getStashBucketURI(rp.domain, rp.title, rp.revision, tid)
+                })
+                .then((res) => Object.assign(
+                    { status: 200 },
+                    JSON.parse(res.body.toString('utf8'))[format])
```
Comment (d00rman) [inline sys/parsoid.js]: It wouldn't be a bad idea to surround this by a try/catch perhaps?

File: sys/parsoid.js [hunk 3]
```diff
@@ -274,15 +290,12 @@ class ParsoidService {
...
-            uri: this.getFallbackBucketURI(rp, 'data-parsoid', tid),
-            headers: parsoidResp.body['data-parsoid'].headers,
-            body: parsoidResp.body['data-parsoid'].body
-        })
-        .then(() => hyper.put({
-            uri: this.getFallbackBucketURI(rp, 'html', tid),
-            headers: parsoidResp.body.html.headers,
-            body: parsoidResp.body.html.body
-        }));
+            uri: this.getStashBucketURI(rp.domain, rp.title, rp.revision, tid),
```
Comment (d00rman) [inline sys/parsoid.js]: Let's add the headers here too

File: sys/parsoid.js (hunk 1 above)
Comment (Pchelolo) [reply sys/parsoid.js]: This is wrong, you are right. THis is not even used in this PR. Shall remove.

File: sys/parsoid.js (hunk 2 above)
Comment (Pchelolo) [reply sys/parsoid.js]: we are already inside a promise, what would we achieve with a try-catch?

File: sys/parsoid.js (hunk 3 above)
Comment (Pchelolo) [reply sys/parsoid.js]: Indeed.

File: sys/parsoid.js (hunk 2 above)
Comment (d00rman) [reply sys/parsoid.js]: A nice error msg. But this was just an observation, we can include it later

**Pull request merged**